### ISSUE TITLE
Client syncs unencrypted 

### ISSUE THREAD

**Issue by FetchFast** [syncthing/syncthing#8974]
Although this situation is made common due to the out of date packages in Ubuntu, _the way it is handled_ by newer syncthing clients is dangerous. 

When an new client connects to an old client without encryption support(i.e. v1.13), and encryption / untrusted client is selected, the data is sent unencrypted. Later, only if the user is present to see it, the mismatch between encrypted and unencrypted will be shown to the user. 

What I'd expect:

I'd like to see a button to press to check that the receiving client can do encrypted storage before the user can select "untrusted device" features. 

This needs to be fixed, not just because of old clients, but as a safety mechanism for the unexpected.

**Comment by calmh** [comment]
Hi! You'll have to be more specific with reproduction steps, because I don't understand the issue. I tried to reproduce it by taking a current Syncthing, setting a peer device as untrusted and sharing a folder with a password, then setting up that peer device as Syncthing 1.11.0 downloaded from the archive (1.13 actually does support encryption, just not in the GUI). What happens is that the connection is closed due to the config error. On the modern side:

```
[I6KAH] 2023/07/09 08:59:58 INFO: Established secure connection to MRIW7OK-NETT3M4-N6SBWME-N25O76W-YJKVXPH-FUMQJ3S-P57B74J-GBITBAC at 127.0.0.1:51498-127.0.0.1:22002/tcp-client/TLS1.3-TLS_AES_128_GCM_SHA256/LAN-P10
[I6KAH] 2023/07/09 08:59:58 INFO: Device MRIW7OK-NETT3M4-N6SBWME-N25O76W-YJKVXPH-FUMQJ3S-P57B74J-GBITBAC client is "syncthing v1.11.0" named "s2" at 127.0.0.1:51498-127.0.0.1:22002/tcp-client/TLS1.3-TLS_AES_128_GCM_SHA256/LAN-P10
[I6KAH] 2023/07/09 08:59:58 INFO: Connection to MRIW7OK-NETT3M4-N6SBWME-N25O76W-YJKVXPH-FUMQJ3S-P57B74J-GBITBAC at 127.0.0.1:51498-127.0.0.1:22002/tcp-client/TLS1.3-TLS_AES_128_GCM_SHA256/LAN-P10 closed: handling cluster-config: remote expects to exchange plain data, but local data is encrypted (folder-type receive-encrypted)
```

and on the outdated side:

```
[MRIW7] 08:59:58 INFO: Established secure connection to I6KAH76-66SLLLB-5PFXSOA-UFJCDZC-YAOMLEK-CP2GB32-BV5RQST-3PSROAU at 127.0.0.1:22002-127.0.0.1:51498/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256
[MRIW7] 08:59:58 INFO: Device I6KAH76-66SLLLB-5PFXSOA-UFJCDZC-YAOMLEK-CP2GB32-BV5RQST-3PSROAU client is "syncthing v1.23.7-dev.14.g25ec2b63.dirty" named "s1" at 127.0.0.1:22002-127.0.0.1:51498/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256
[MRIW7] 08:59:58 INFO: Connection to I6KAH76-66SLLLB-5PFXSOA-UFJCDZC-YAOMLEK-CP2GB32-BV5RQST-3PSROAU at 127.0.0.1:22002-127.0.0.1:51498/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256 closed: reading length: read tcp 127.0.0.1:22002->127.0.0.1:51498: read: connection reset by peer
```

It doesn't have a smart error message because it predates encryption. (1.13.0 correctly explains the mismatch)

**Comment by FetchFast** [comment]
Thanks for investigating! To be honest, it's too much for me to investigate further at this time. I think it's best to close this one out, but leave it discoverable in case anyone finds the same thing.
